From 256d1e0e5841b1f25042fc41c64742a02b2e54d4 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sun, 19 Sep 2021 19:00:12 +0000
Subject: [PATCH 20/52] Revert "Disable FOD animation if animation resource
 package not installed"

This reverts commit a41a8ecd063e4f6fdfea9f61ba6609d32bc0569f.
---
 core/res/res/values/cr_config.xml             |  3 --
 core/res/res/values/cr_symbols.xml            |  3 --
 .../systemui/biometrics/FODAnimation.java     |  8 ++---
 .../systemui/biometrics/FODCircleView.java    | 33 +++++--------------
 4 files changed, 13 insertions(+), 34 deletions(-)

diff --git a/core/res/res/values/cr_config.xml b/core/res/res/values/cr_config.xml
index 6afe79a3f75..2ae48a08f5e 100644
--- a/core/res/res/values/cr_config.xml
+++ b/core/res/res/values/cr_config.xml
@@ -71,9 +71,6 @@
     <!-- Default fod pressed color -->
     <integer name="config_fod_pressed_color">1</integer>
 
-    <!-- Default value for FOD animation package resources -->
-    <string name="config_fodAnimationPackage" translatable="false">com.crdroid.fod.animations</string>
-
     <!-- Whether device has physical tri state switch -->
     <bool name="config_hasAlertSlider">false</bool>
 
diff --git a/core/res/res/values/cr_symbols.xml b/core/res/res/values/cr_symbols.xml
index 95b8b044e59..f0dda389835 100644
--- a/core/res/res/values/cr_symbols.xml
+++ b/core/res/res/values/cr_symbols.xml
@@ -80,9 +80,6 @@
   <!-- Default fod pressed icon -->
   <java-symbol type="integer" name="config_fod_pressed_color" />
 
-  <!-- Default value for FOD animation package resources -->
-  <java-symbol type="string" name="config_fodAnimationPackage" />
-
   <!-- App error dialog -->
   <java-symbol type="id" name="aerr_copy" />
   <java-symbol type="string" name="url_copy_success" />
diff --git a/packages/SystemUI/src/com/android/systemui/biometrics/FODAnimation.java b/packages/SystemUI/src/com/android/systemui/biometrics/FODAnimation.java
index 642f73785b9..67b92af5889 100644
--- a/packages/SystemUI/src/com/android/systemui/biometrics/FODAnimation.java
+++ b/packages/SystemUI/src/com/android/systemui/biometrics/FODAnimation.java
@@ -71,7 +71,7 @@ public class FODAnimation extends ImageView {
         "fod_rog_supernova_recognizing_anim",
     };
 
-    private final String mFodAnimationPackage;
+    private final String FOD_ANIMATIONS_PACKAGE = "com.crdroid.fod.animations";
 
     private static final boolean DEBUG = true;
     private static final String LOG_TAG = "FODAnimations";
@@ -81,7 +81,7 @@ public class FODAnimation extends ImageView {
 
         mContext = context;
         mWindowManager = mContext.getSystemService(WindowManager.class);
-        mFodAnimationPackage = mContext.getResources().getString(com.android.internal.R.string.config_fodAnimationPackage);
+
         mAnimationSize = mContext.getResources().getDimensionPixelSize(R.dimen.fod_animation_size);
         mAnimationOffset = mContext.getResources().getDimensionPixelSize(R.dimen.fod_animation_offset);
         mAnimParams.height = mAnimationSize;
@@ -107,8 +107,8 @@ public class FODAnimation extends ImageView {
         int resId = 0;
         try {
             PackageManager pm = mContext.getPackageManager();
-            Resources mApkResources = pm.getResourcesForApplication(mFodAnimationPackage);
-            resId = mApkResources.getIdentifier(drawableName, "drawable", mFodAnimationPackage);
+            Resources mApkResources = pm.getResourcesForApplication(FOD_ANIMATIONS_PACKAGE);
+            resId = mApkResources.getIdentifier(drawableName, "drawable", FOD_ANIMATIONS_PACKAGE);
             if (DEBUG) Log.i(LOG_TAG, "Got resource id: "+ resId +" from package" );
             setBackgroundDrawable(mApkResources.getDrawable(resId));
             recognizingAnim = (AnimationDrawable) getBackground();
diff --git a/packages/SystemUI/src/com/android/systemui/biometrics/FODCircleView.java b/packages/SystemUI/src/com/android/systemui/biometrics/FODCircleView.java
index fb5e3971d9d..b6452783ecc 100644
--- a/packages/SystemUI/src/com/android/systemui/biometrics/FODCircleView.java
+++ b/packages/SystemUI/src/com/android/systemui/biometrics/FODCircleView.java
@@ -48,7 +48,6 @@ import android.view.View;
 import android.view.WindowManager;
 import android.widget.ImageView;
 
-import com.android.internal.util.crdroid.Utils;
 import com.android.internal.widget.LockPatternUtils;
 import com.android.keyguard.KeyguardSecurityModel.SecurityMode;
 import com.android.keyguard.KeyguardUpdateMonitor;
@@ -111,7 +110,6 @@ public class FODCircleView extends ImageView implements TunerService.Tunable {
 
     private FODAnimation mFODAnimation;
     private boolean mIsRecognizingAnimEnabled;
-    private boolean mIsFodAnimationAvailable = false;
 
     private int mSelectedIcon;
     private final int[] ICON_STYLES = {
@@ -224,7 +222,7 @@ public class FODCircleView extends ImageView implements TunerService.Tunable {
         public void onKeyguardVisibilityChanged(boolean showing) {
             mIsKeyguard = showing;
             updateStyle();
-            if (mIsFodAnimationAvailable && mFODAnimation != null) {
+            if (mFODAnimation != null) {
                 mFODAnimation.setAnimationKeyguard(mIsKeyguard);
             }
         }
@@ -242,7 +240,7 @@ public class FODCircleView extends ImageView implements TunerService.Tunable {
             } else {
                 hide();
             }
-            if (mIsFodAnimationAvailable && mFODAnimation != null) {
+            if (mFODAnimation != null) {
                 mFODAnimation.setAnimationKeyguard(mIsBouncer);
             }
         }
@@ -279,7 +277,7 @@ public class FODCircleView extends ImageView implements TunerService.Tunable {
         @Override
         public void onBiometricHelp(int msgId, String helpString,
                 BiometricSourceType biometricSourceType) {
-            if (msgId == -1 && mIsFodAnimationAvailable) { // Auth error
+            if (msgId == -1){ // Auth error
                 mHandler.post(() -> mFODAnimation.hideFODanimation());
             }
         }
@@ -362,12 +360,7 @@ public class FODCircleView extends ImageView implements TunerService.Tunable {
         mUpdateMonitor = Dependency.get(KeyguardUpdateMonitor.class);
         mUpdateMonitor.registerCallback(mMonitorCallback);
 
-        mIsFodAnimationAvailable = Utils.isPackageInstalled(context,
-                                    context.getResources().getString(
-                                    com.android.internal.R.string.config_fodAnimationPackage));
-        if (mIsFodAnimationAvailable) {
-            mFODAnimation = new FODAnimation(context, mPositionX, mPositionY);
-        }
+        mFODAnimation = new FODAnimation(context, mPositionX, mPositionY);
 
         Dependency.get(TunerService.class).addTunable(this, FOD_GESTURE,
                 Settings.Secure.DOZE_ENABLED);
@@ -447,9 +440,7 @@ public class FODCircleView extends ImageView implements TunerService.Tunable {
             return true;
         }
 
-        if (mIsFodAnimationAvailable) {
-            mHandler.post(() -> mFODAnimation.hideFODanimation());
-        }
+        mHandler.post(() -> mFODAnimation.hideFODanimation());
         return false;
     }
 
@@ -522,9 +513,7 @@ public class FODCircleView extends ImageView implements TunerService.Tunable {
         setDim(true);
         dispatchPress();
 
-        if (mIsFodAnimationAvailable) {
-            mHandler.post(() -> mFODAnimation.showFODanimation());
-        }
+        mHandler.post(() -> mFODAnimation.showFODanimation());
 
         setImageDrawable(null);
         invalidate();
@@ -539,9 +528,7 @@ public class FODCircleView extends ImageView implements TunerService.Tunable {
         dispatchRelease();
         setDim(false);
 
-        if (mIsFodAnimationAvailable) {
-            mHandler.post(() -> mFODAnimation.hideFODanimation());
-        }
+        mHandler.post(() -> mFODAnimation.hideFODanimation());
 
         setKeepScreenOn(false);
     }
@@ -617,7 +604,7 @@ public class FODCircleView extends ImageView implements TunerService.Tunable {
         mPressedColor = Settings.System.getInt(mContext.getContentResolver(),
                 Settings.System.FOD_COLOR, mDefaultPressedColor);
 
-        if (mIsFodAnimationAvailable && mFODAnimation != null) {
+        if (mFODAnimation != null) {
             mFODAnimation.update(mIsRecognizingAnimEnabled);
         }
     }
@@ -657,9 +644,7 @@ public class FODCircleView extends ImageView implements TunerService.Tunable {
         if (mIsDreaming) {
             mParams.x += mDreamingOffsetX;
             mParams.y += mDreamingOffsetY;
-            if (mIsFodAnimationAvailable) {
-                mFODAnimation.updateParams(mParams.y);
-            }
+            mFODAnimation.updateParams(mParams.y);
         }
 
         mWindowManager.updateViewLayout(this, mParams);
-- 
2.30.2

